breacharbiter.go

```diff
@@ -911,31 +941,51 @@ func newRetributionInfo(chanPoint *wire.OutPoint,
 	breachInfo *lnwallet.BreachRetribution,
 	chanInfo *channeldb.ChannelSnapshot) *retributionInfo {
 
-	// First, record the breach information and witness type for the local
-	// channel point.  This will allow us to completely generate a valid
-	// witness in the event of failures, as it will be persisted in the
-	// retribution store.  Here we use CommitmentNoDelay since this output
-	// belongs to us and has no time-based constraints on spending.
-	selfOutput := newBreachedOutput(&breachInfo.LocalOutpoint,
-		lnwallet.CommitmentNoDelay, &breachInfo.LocalOutputSignDesc)
-
-	// Second, record the same information and witness type regarding the
-	// remote outpoint, which belongs to the party who tried to steal our
-	// money! Here we set witnessType of the breachedOutput to
-	// CommitmentRevoke, since we will be using a revoke key, withdrawing
-	// the funds from the commitment transaction immediately.
-	revokedOutput := newBreachedOutput(&breachInfo.RemoteOutpoint,
-		lnwallet.CommitmentRevoke, &breachInfo.RemoteOutputSignDesc)
-
 	// Determine the number of second layer HTLCs we will attempt to sweep.
 	nHtlcs := len(breachInfo.HtlcRetributions)
 
-	// Lastly, for each of the breached HTLC outputs, assemble the
-	// information we will persist to disk, such that we will be able to
-	// deterministically generate a valid witness for each output. This will
-	// allow the breach arbiter to recover from failures, in the event that
-	// it must sign and broadcast the justice transaction.
-	htlcOutputs := make([]*breachedOutput, nHtlcs)
+	// Initialize a slice to hold the outputs we will attempt to sweep. The
+	// maximum capacity of the slice is set to 2+nHtlcs to handle the case
+	// where the local, remote, and all HTLCs are not dust outputs.  All
+	// HTLC outputs provided by the wallet are guaranteed to be non-dust,
+	// though the commitment outputs are conditionally added depending on
+	// the nil-ness of their sign descriptors.
+	breachedOutputs := make([]breachedOutput, nHtlcs+2)
```

Comment:
Seems like we could drop the usage of the `offset` variable all together by changing this to be: 
```
breachedOutputs := make([]breachedOutput, 0, nHtlcs+2)
```

And then using `append` below.

Reply:
Fixed! 🎉